### ISSUE TITLE
Pausing comparing doesn't pause immediately

### ISSUE THREAD

**Issue by JeroenOortwijn** [WinMerge/winmerge#342]
When comparing two folders with large files (>50 GB) on a relatively slow medium (HD), WinMerge 2.16.6.0 doesn't immediately pause the comparison when you click on the Pause button.

Steps to reproduce:
- Start comparing two identical folders recursively with large files (>50 GB) in them.
- When WinMerge starts comparing a new file, press the Pause button.
- Notice in the Task Manager that WinMerge is still using the disk.
- Notice that as soon WinMerge is done comparing this file, it now pauses.

Preferred behaviour:
When pressing the Pause button, WinMerge should immediately pause the comparison and don't use the disk.

**Comment by pbucolo** [comment]
I've got the same issue, but I noticed a little bit different behavior.

Common condition is to compare identical folders with a large NUMBER of files (my test was made with more then  a million files).

- When Stop (or Pause) button is clicked and Winmerge is comparing a large file (in my test I clicked when WM compare a file with size more then 8 GB) I note that WM freezes until the comparison ends.

- Pierluigi
